runtime/metricsview/executor_timestamps.go

```diff
@@ -0,0 +1,243 @@
+package metricsview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	defaultExecutionTimeout = time.Minute * 3
+)
+
+func (e *Executor) resolveDuckDB(ctx context.Context) (time.Time, time.Time, time.Time, error) {
+	timeDim := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	escapedTableName := e.olap.Dialect().EscapeTable(e.metricsView.Database, e.metricsView.DatabaseSchema, e.metricsView.Table)
+
+	var watermarkExpr string
+	if e.metricsView.WatermarkExpression != "" {
+		watermarkExpr = e.metricsView.WatermarkExpression
+	} else {
+		watermarkExpr = fmt.Sprintf("MAX(%s)", timeDim)
+	}
+
+	rangeSQL := fmt.Sprintf(
+		"SELECT min(%[1]s) as \"min\", max(%[1]s) as \"max\", %[2]s as \"watermark\", max(%[1]s) - min(%[1]s) as \"interval\" FROM %[3]s %[4]s",
+		timeDim,
+		watermarkExpr,
+		escapedTableName,
+		e.security.RowFilter(),
+	)
+
+	rows, err := e.olap.Execute(ctx, &drivers.Statement{
+		Query:            rangeSQL,
+		Priority:         e.priority,
+		ExecutionTimeout: defaultExecutionTimeout,
+	})
+	if err != nil {
+		return time.Time{}, time.Time{}, time.Time{}, err
+	}
+	defer rows.Close()
+
+	if rows.Next() {
+		rowMap := make(map[string]any)
+		err = rows.MapScan(rowMap)
+		if err != nil {
+			return time.Time{}, time.Time{}, time.Time{}, err
+		}
+		if v := rowMap["min"]; v != nil {
+			minTime, ok := v.(time.Time)
+			if !ok {
+				return time.Time{}, time.Time{}, time.Time{}, fmt.Errorf("not a timestamp column")
+			}
+			return minTime, rowMap["max"].(time.Time), rowMap["watermark"].(time.Time), nil
+		}
```

Comment:
I think it would be simpler to do:
```go
var minTime, maxTime, watermark time.Time
...
err = rows.Scan(&minTime, &maxTime, &watermark)
```

`Scan` will then handle nulls, casting it to a `time.Time`, etc.